charts/operator/chart_test.go

```diff
@@ -1,28 +1,185 @@
+	// redpanda helm chart.
+	out, err := exec.Command("helm", "repo", "add", "prometheus", "https://prometheus-community.github.io/helm-charts").CombinedOutput()
+	if err != nil {
+		log.Fatalf("failed to run helm repo add: %s", out)
+	}
+
+	out, err = exec.Command("helm", "dep", "build", ".").CombinedOutput()
+	if err != nil {
+		log.Fatalf("failed to run helm dep build: %s", out)
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestHelmKustomizeEquivalence(t *testing.T) {
+	ctx := testutil.Context(t)
+	client, err := helm.New(helm.Options{ConfigHome: testutil.TempDir(t)})
+	require.NoError(t, err)
+
+	values := PartialValues{FullnameOverride: ptr.To("redpanda")}
+
+	rendered, err := client.Template(ctx, ".", helm.TemplateOptions{
+		Name:      "redpanda",
+		Namespace: "",
+		Values:    values,
+	})
+	require.NoError(t, err)
+
+	fSys := filesys.MakeFsOnDisk()
+	buffy := new(bytes.Buffer)
+	cmd := build.NewCmdBuild(
+		fSys, build.MakeHelp(konfig.ProgramName, "build"), buffy)
+	require.NoError(t, cmd.RunE(cmd, []string{"https://github.com/redpanda-data/redpanda-operator//operator/config/default"}))
+
+	helmObjs, err := kube.DecodeYAML(rendered, Scheme)
+	require.NoError(t, err)
+
+	require.NoError(t, apiextensionsv1.AddToScheme(Scheme))
+	kustomizeObjs, err := kube.DecodeYAML(buffy.Bytes(), Scheme)
+	require.NoError(t, err)
+
+	helmClusterRoleRules, helmRoleRules := ExtractRules(helmObjs)
+	kClusterRoleRules, kRoleRules := ExtractRules(kustomizeObjs)
+
+	// Special 2 cases as namespaced operator deployment have only Role when cluster scoped deployment has
+	// ClusterRole rules
+	clusterKey := "redpanda.vectorized.io#clusters"
+	consoleKey := "redpanda.vectorized.io#consoles"
+
+	helmChart := helmRoleRules[clusterKey]
+	// As namespaced operator deployment does not have create and delete permissions for Cluster resource
+	helmChart = append(helmChart, "create", "delete")
+	slices.Sort(helmChart)
+	kustomize := kClusterRoleRules[clusterKey]
+	slices.Sort(kustomize)
+	assert.Equal(t, helmChart, kustomize, "for the api#resource %q", clusterKey)
+	// Delete cluster resource key as default helm base deployment only is defined only
+	// in Role when in kustomize the same Resource is defined in ClusterRole
+	delete(helmRoleRules, clusterKey)
+	delete(kClusterRoleRules, clusterKey)
+
+	helmChart = helmRoleRules[consoleKey]
+	// As namespaced operator deployment does not have create and delete permissions for Console resource
+	helmChart = append(helmChart, "create", "delete")
+	slices.Sort(helmChart)
+	kustomize = kClusterRoleRules[consoleKey]
+	slices.Sort(kustomize)
+	assert.Equal(t, helmChart, kustomize, "for the api#resource %q", consoleKey)
+	// Delete console resource key as default helm base deployment only is defined only
+	// in Role when in kustomize the same Resource is defined in ClusterRole
+	delete(helmRoleRules, consoleKey)
+	delete(kClusterRoleRules, consoleKey)
+
+	for k := range helmRoleRules {
```

Comment:
This to me says we should probably do something to make the permissions between the chart and the kustomize manifest much more similar 😓 

Do you think it would help if we [ran `controller-gen` in the operator](https://github.com/redpanda-data/redpanda-operator/blob/c224a04dc20c8d51e6e6543a8dd45b5f03b1e7e5/taskfiles/k8s.yml#L21) separately for V1 and V2 so we'd have distinct sets that we can individually run kustomize on or merge via the `default` folder?

Reply:
That would be ideal! Should I do it right away and align this PR?

Reply:
If you don't think it'll be too much extra effort, I'd be in favor of it!